src/OutputAreaViewModel.ts

```diff
@@ -0,0 +1,374 @@
+    return OutputAreaViewModel.stateChangedSignal.bind(this);
+  }
+
+  /**
+   * Get whether the output has a maximum fixed height.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[fixedHeightProperty]].
+   */
+  get fixedHeight() {
+    return OutputAreaViewModel.fixedHeightProperty.get(this);
+  }
+
+  /**
+   * Set whether the output has a maximum fixed height.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[fixedHeightProperty]].
+   */
+  set fixedHeight(value: boolean) {
+    OutputAreaViewModel.fixedHeightProperty.set(this, value);
+  }
+
+  /**
+   * Get whether the input area should be collapsed or displayed.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[collapsedProperty]].
+   */
+  get collapsed() {
+    return OutputAreaViewModel.collapsedProperty.get(this);
+  }
+
+  /**
+   * Set whether the input area should be collapsed or displayed.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[collapsedProperty]].
+   */
+  set collapsed(value: boolean) {
+    OutputAreaViewModel.collapsedProperty.set(this, value);
+  }
+
+  /**
+   * Get the prompt.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[promptProperty]].
+   */
+  get prompt() {
+    return OutputAreaViewModel.promptProperty.get(this);
+  }
+
+  /**
+   * Set the prompt.
+   *
+   * #### Notes
+   * This is a pure delegate to the [[promptProperty]].
+   */
+  set prompt(value: string) {
+    OutputAreaViewModel.promptProperty.set(this, value);
+  }
+  
+  /**
+   * Add an output, which may be combined with previous output 
+   * (e.g. for streams).
+   */
+  add(output: OutputViewModel) {
+    // if we received a delayed clear message, then clear now
+    if (this._clearNext) {
+      this.clear();
+      this._clearNext = false;
+    }
+    
+    // Consolidate outputs if they are stream outputs of the same kind
+    let lastOutput = this.outputs.get(-1);
+    if (isStreamViewModel(output)
+        && isStreamViewModel(lastOutput)
+        && output.name === lastOutput.name) {
+      // In order to get a list change event, we add the previous 
+      // text to the current item and replace the previous item. 
+      // This also replaces the metadata of the last item.
+      output.text = lastOutput.text + output.text;
+      this.outputs.set(-1, output);
+    } else {
+      this.outputs.add(output);
+    }
+  }
+
+  /**
+  * Clear all of the output.
+  * 
+  * @param wait Delay clearing the output until the next message is added.
+  */
+  clear(wait: boolean = false) {
+    if(wait) {
+      this._clearNext = true;
+    } else {
+      this.outputs.clear();
+    }
+  }
+
+  outputs = new ObservableList<OutputViewModel>();
+  
+  /**
+   * Whether to clear on the next message add.
+   */
+  private _clearNext = false;
+}
+
+
+/**
+  * A function to update an output area viewmodel to reflect a stream of messages 
+  */
+export
+function consumeMessage(msg: any, outputArea: OutputAreaViewModel): void {
```

Comment:
Should this be `IOutputAreaViewModel` if your intent is for this to be generic?  Otherwise, it should be a method of `OutputAreaViewModel`.